File: ClientApp/src/app/models/search-params.ts
import { HeaderType } from "./header-type";
import { PaginationParams } from "./pagination-params";

export interface SearchParams extends PaginationParams {
    searchTerm?: string;
    headerType?: HeaderType
    date?: Date;
    sort?: string;
    latitude?: number;
    longitude?: number;
    radiusKm?: number;
    genreIds?: number[];
}

File: Core/Parameters/SearchParams.cs
using Core.Interfaces;

namespace Core.Parameters
{
    public class SearchParams : IPageParams, IGeoParams
    {
        public int PageNumber { get; set; } = 1;
        public int PageSize { get; set; } = 10;
        public string? SearchTerm { get; set; }
        public string HeaderType { get; set; }
        public DateTime? Date { get; set; }
        public string? Sort { get; set; }
        public double? Latitude { get; set; }
        public double? Longitude { get; set; }
        public int? RadiusKm { get; set; }
        public int[]? GenreIds { get; set; }
        public bool? ShowHistory { get; set; }
        public bool? ShowSold { get; set; }
    }
}
